server/fleet/queries_test.go

```diff
@@ -180,3 +180,32 @@ func TestRoundtripQueriesYaml(t *testing.T) {
 		})
 	}
 }
+
+func TestVerifyQueryPlatforms(t *testing.T) {
+	testCases := []struct {
+		name           string
+		platformString string
+		shouldErr      bool
+	}{
+		{"empty platform string okay", "", false},
+		{"platform string 'darwin' okay", "darwin", false},
+		{"platform string 'linux' okay", "linux", false},
+		{"platform string 'windows' okay", "windows", false},
+		{"platform string 'darwin,linux,windows' okay", "darwin,linux,windows", false},
+		{"platform string 'foo' invalid – not a supported platform", "foo", true},
+		{"platform string 'charles,darwin,linux,windows' invalid – 'charles' not a supported platform", "charles,darwin,linux,windows", true},
+		{"platform string 'darwin windows' invalid – missing comma delimiter", "darwin windows", true},
+		{"platform string 'charles darwin' invalid – 'charles' not supported and missing comma delimiter", "charles darwin", true},
```

Comment:
nice coverage!  any need for trailing whitespace or special characters?

Reply:
Good idea, I'll add some special character cases. Re trailing whitespace, I think that falls under [this conversation above](https://github.com/fleetdm/fleet/pull/13923#discussion_r1325881963) – TL;DR, the analagous test for policies is trimming whitespace, so for now we're following that pattern. May choose to disallow extra whitespace in the future.